analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CollectionEmptinessChecking.Fixed.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Tests.Diagnostics
+{
+    public class CollectionEmptinessChecking
+    {
+        private static bool HasContent1(IEnumerable<string> l)
+        {
+            return l.Any; // Fixed
+        }
+        private static bool HasContent1b(IEnumerable<string> l)
+        {
+            return l.Any; // Fixed
+        }
+        private static bool HasContent2(List<string> l)
+        {
+            return Enumerable.Any; // Fixed
+        }
+        private static bool HasContent2b(List<string> l)
+        {
+            return Enumerable.Any; // Fixed
+        }
+        private static bool IsNotEmpty1(List<string> l)
+        {
+            return l.Any; // Fixed
+        }
+        private static bool IsNotEmpty2(List<string> l)
+        {
+            return l.Any; // Fixed
+        }
+        private static bool IsEmpty1(List<string> l)
+        {
+            return !l.Any; // Fixed
+        }
+        private static bool IsEmpty2(List<string> l)
+        {
+            return !l.Any; // Fixed
+        }
+        private static bool IsEmpty2b(List<string> l)
+        {
+            return !l.Any; // Fixed
+        }
+        private static bool IsEmpty4(List<string> l)
+        {
+            return !l.Any; // Fixed
+        }
+        private static bool IsEmpty4b(List<string> l)
+        {
+            return !l.Any; // Fixed
+        }
+        private static bool HasContentWithCondition(List<int> numbers)
+        {
+            return numbers.Any; // Fixed
```

Comment:
This isn't the correct fix. We need to preserve the linq condition to  achieve the same result.
```suggestion
            return numbers.Any(n => n % 2 == 0); // Fixed
```